pkg/pillar/zedcloud/deferred.go

```diff
@@ -66,55 +85,90 @@ func HandleDeferred(zedcloudCtx *ZedCloudContext, event time.Time, spacing time.
 
 func (ctx *DeferredContext) handleDeferred(log *base.LogObject, event time.Time,
 	spacing time.Duration) bool {
+	ctx.lock.Lock()
+	defer ctx.lock.Unlock()
 
-	log.Functionf("HandleDeferred(%v, %v) map %d\n",
+	if len(ctx.deferredItems) == 0 {
+		return true
+	}
+
+	log.Functionf("HandleDeferred(%v, %v) items %d\n",
 		event, spacing, len(ctx.deferredItems))
-	iteration := 0 // Do some load spreading
-	for key, l := range ctx.deferredItems {
-		log.Functionf("Trying to send for %s items %d\n", key, len(l.list))
-		failed := false
-		for i, item := range l.list {
-			if item.buf == nil {
-				continue
-			}
-			if item.buf.Len() == 0 {
-				log.Errorf("Zero length defered item for %s",
-					key)
-				continue
+
+	//sort items by added time in ascending order
+	sort.Slice(ctx.deferredItems, func(i, j int) bool {
+		t1 := time.Time{}
+		for _, el := range ctx.deferredItems[i].list {
```

Comment:
It we remove the unused addDeferred() function we will no longer need a list here and things will be bit simpler.